File: pkg/dcdflib/src/vec_cdf.c
/* ------------------------------------------------------------

 vectorized version of DCDFLIB  cdf**** functions.

 Use the same arguments as cdf***,
 just an additional int *len : simply loop through them *len times

 --- following the idea of  Donald H. MacQueen <macqueen1@llnl.gov>
 --- who did this for cdftnc only

 © 1999--2018 Martin Maechler
 This is GNU Public Licensed Software.
 ------------------------------------------------------------
*/

#include "vec_cdf.h"

void V_cdfbet(int *which, double *p, double *q, double *x, double *y,
	      double *a, double *b, int *status, double *bound, int *len)
{
    int i;
    for (i = 0; i < *len; i++) {
	cdfbet((int *)which, &p[i], &q[i], &x[i], &y[i],
	       &a[i], &b[i], (int *)&status[i], &bound[i]);
    }
}


void V_cdfbin(int *which, double *p, double *q, double *s, double *xn,
	      double *pr, double *ompr, int *status, double *bound, int *len)
{
    int i;
    for (i = 0; i < *len; i++) {
	cdfbin((int *)which, &p[i], &q[i], &s[i], &xn[i],
	       &pr[i], &ompr[i], (int *)&status[i], &bound[i]);
    }
}


void V_cdfchi(int *which, double *p, double *q, double *x, double *df,
	      int *status, double *bound, int *len)
{
    int i;
    for (i = 0; i < *len; i++) {
	cdfchi((int *)which, &p[i], &q[i], &x[i], &df[i],
	       (int *)&status[i], &bound[i]);
    }
}

void V_cdfchn(int *which, double *p, double *q, double *x, double *df,
	      double *pnonc, int *status, double *bound, int *len)
{
    int i;
    for (i = 0; i < *len; i++) {
	cdfchn((int *)which, &p[i], &q[i], &x[i], &df[i],
	       &pnonc[i], (int *)&status[i], &bound[i]);
    }
}


void V_cdff(int *which, double *p, double *q, double *f, double *dfn,
	    double *dfd, int *status, double *bound, int *len)
{
    int i;
    for (i = 0; i < *len; i++) {
	cdff((int *)which, &p[i], &q[i], &f[i], &dfn[i],
	     &dfd[i], (int *)&status[i], &bound[i]);
    }
}

void V_cdffnc(int *which, double *p, double *q, double *f, double *dfn,
	      double *dfd, double *phonc, int *status, double *bound, int *len)
{
    int i;
    for (i = 0; i < *len; i++) {
	cdffnc((int *)which, &p[i], &q[i], &f[i], &dfn[i],
	       &dfd[i], &phonc[i], (int *)&status[i], &bound[i]);
    }
}


void V_cdfgam(int *which, double *p, double *q, double *x, double *shape,
	      double *scale, int *status, double *bound, int *len)
{
    int i;
    for (i = 0; i < *len; i++) {
	cdfgam((int *)which, &p[i], &q[i], &x[i], &shape[i],
	       &scale[i], (int *)&status[i], &bound[i]);
    }
}


void V_cdfnbn(int *which, double *p, double *q, double *s, double *xn,
	      double *pr, double *ompr, int *status, double *bound, int *len)
{
    int i;
    for (i = 0; i < *len; i++) {
	cdfnbn((int *)which, &p[i], &q[i], &s[i], &xn[i],
	       &pr[i], &ompr[i], (int *)&status[i], &bound[i]);
    }
}


void V_cdfnor(int *which, double *p, double *q, double *x, double *mean,
	      double *sd, int *status, double *bound, int *len)
{
    int i;
    for (i = 0; i < *len; i++) {
	cdfnor((int *)which, &p[i], &q[i], &x[i], &mean[i],
	       &sd[i], (int *)&status[i], &bound[i]);
    }
}


void V_cdfpoi(int *which, double *p, double *q, double *s, double *xlam,
	      int *status, double *bound, int *len)
{
    int i;
    for (i = 0; i < *len; i++) {
	cdfpoi((int *)which, &p[i], &q[i], &s[i], &xlam[i],
	       (int *)&status[i], &bound[i]);
    }
}


void V_cdft(int *which, double *p, double *q, double *t, double *df,
	    int *status, double *bound, int *len)
{
    int i;
    for (i = 0; i < *len; i++) {
	cdft((int *)which, &p[i], &q[i], &t[i], &df[i],
	     (int *)&status[i], &bound[i]);
    }
}

void V_cdftnc(int *which, double *p, double *q, double *t, double *df,
	      double *pnonc, int *status, double *bound, int *len)
{
    int i;
    for (i = 0; i < *len; i++) {
	cdftnc((int *)which, &p[i], &q[i], &t[i], &df[i],
	       &pnonc[i], (int *)&status[i], &bound[i]);
    }
}

File: pkg/dcdflib/R/vec_cdf.R
#### Automagically produced by
###  cd ../src
'
 ./C2R vec_cdf.c | ./R2VR > vec_cdf.R
'

cdfbetV <- function(which, p, q, x, y, a, b, status, bound)
{
  len <- max(length(p), length(q), length(x), length(y), length(a), length(b), length(status), length(bound))
 .C("V_cdfbet",
        which = as.integer(which),
            p = rep(as.double(p), length = len),
            q = rep(as.double(q), length = len),
            x = rep(as.double(x), length = len),
            y = rep(as.double(y), length = len),
            a = rep(as.double(a), length = len),
            b = rep(as.double(b), length = len),
       status = rep(as.integer(status), length = len),
        bound = rep(as.double(bound), length = len),
          len = as.integer(len)
  , PACKAGE = "dcdflib")
}

cdfbinV <- function(which, p, q, s, xn, pr, ompr, status, bound)
{
  len <- max(length(p), length(q), length(s), length(xn), length(pr), length(ompr), length(status), length(bound))
 .C("V_cdfbin",
        which = as.integer(which),
            p = rep(as.double(p), length = len),
            q = rep(as.double(q), length = len),
            s = rep(as.double(s), length = len),
           xn = rep(as.double(xn), length = len),
           pr = rep(as.double(pr), length = len),
         ompr = rep(as.double(ompr), length = len),
       status = rep(as.integer(status), length = len),
        bound = rep(as.double(bound), length = len),
          len = as.integer(len)
  , PACKAGE = "dcdflib")
}

cdfchiV <- function(which, p, q, x, df, status, bound)
{
  len <- max(length(p), length(q), length(x), length(df), length(status), length(bound))
 .C("V_cdfchi",
        which = as.integer(which),
            p = rep(as.double(p), length = len),
            q = rep(as.double(q), length = len),
            x = rep(as.double(x), length = len),
           df = rep(as.double(df), length = len),
       status = rep(as.integer(status), length = len),
        bound = rep(as.double(bound), length = len),
          len = as.integer(len)
  , PACKAGE = "dcdflib")
}

cdfchnV <- function(which, p, q, x, df, pnonc, status, bound)
{
  len <- max(length(p), length(q), length(x), length(df), length(pnonc), length(status), length(bound))
 .C("V_cdfchn",
        which = as.integer(which),
            p = rep(as.double(p), length = len),
            q = rep(as.double(q), length = len),
            x = rep(as.double(x), length = len),
           df = rep(as.double(df), length = len),
        pnonc = rep(as.double(pnonc), length = len),
       status = rep(as.integer(status), length = len),
        bound = rep(as.double(bound), length = len),
          len = as.integer(len)
  , PACKAGE = "dcdflib")
}

cdffV <- function(which, p, q, f, dfn, dfd, status, bound)
{
  len <- max(length(p), length(q), length(f), length(dfn), length(dfd), length(status), length(bound))
 .C("V_cdff",
        which = as.integer(which),
            p = rep(as.double(p), length = len),
            q = rep(as.double(q), length = len),
            f = rep(as.double(f), length = len),
          dfn = rep(as.double(dfn), length = len),
          dfd = rep(as.double(dfd), length = len),
       status = rep(as.integer(status), length = len),
        bound = rep(as.double(bound), length = len),
          len = as.integer(len)
  , PACKAGE = "dcdflib")
}

cdffncV <- function(which, p, q, f, dfn, dfd, phonc, status, bound)
{
  len <- max(length(p), length(q), length(f), length(dfn), length(dfd), length(phonc), length(status), length(bound))
 .C("V_cdffnc",
        which = as.integer(which),
            p = rep(as.double(p), length = len),
            q = rep(as.double(q), length = len),
            f = rep(as.double(f), length = len),
          dfn = rep(as.double(dfn), length = len),
          dfd = rep(as.double(dfd), length = len),
        phonc = rep(as.double(phonc), length = len),
       status = rep(as.integer(status), length = len),
        bound = rep(as.double(bound), length = len),
          len = as.integer(len)
  , PACKAGE = "dcdflib")
}

cdfgamV <- function(which, p, q, x, shape, scale, status, bound)
{
  len <- max(length(p), length(q), length(x), length(shape), length(scale), length(status), length(bound))
 .C("V_cdfgam",
        which = as.integer(which),
            p = rep(as.double(p), length = len),
            q = rep(as.double(q), length = len),
            x = rep(as.double(x), length = len),
        shape = rep(as.double(shape), length = len),
        scale = rep(as.double(scale), length = len),
       status = rep(as.integer(status), length = len),
        bound = rep(as.double(bound), length = len),
          len = as.integer(len)
  , PACKAGE = "dcdflib")
}

cdfnbnV <- function(which, p, q, s, xn, pr, ompr, status, bound)
{
  len <- max(length(p), length(q), length(s), length(xn), length(pr), length(ompr), length(status), length(bound))
 .C("V_cdfnbn",
        which = as.integer(which),
            p = rep(as.double(p), length = len),
            q = rep(as.double(q), length = len),
            s = rep(as.double(s), length = len),
           xn = rep(as.double(xn), length = len),
           pr = rep(as.double(pr), length = len),
         ompr = rep(as.double(ompr), length = len),
       status = rep(as.integer(status), length = len),
        bound = rep(as.double(bound), length = len),
          len = as.integer(len)
  , PACKAGE = "dcdflib")
}

cdfnorV <- function(which, p, q, x, mean, sd, status, bound)
{
  len <- max(length(p), length(q), length(x), length(mean), length(sd), length(status), length(bound))
 .C("V_cdfnor",
        which = as.integer(which),
            p = rep(as.double(p), length = len),
            q = rep(as.double(q), length = len),
            x = rep(as.double(x), length = len),
         mean = rep(as.double(mean), length = len),
           sd = rep(as.double(sd), length = len),
       status = rep(as.integer(status), length = len),
        bound = rep(as.double(bound), length = len),
          len = as.integer(len)
  , PACKAGE = "dcdflib")
}

cdfpoiV <- function(which, p, q, s, xlam, status, bound)
{
  len <- max(length(p), length(q), length(s), length(xlam), length(status), length(bound))
 .C("V_cdfpoi",
        which = as.integer(which),
            p = rep(as.double(p), length = len),
            q = rep(as.double(q), length = len),
            s = rep(as.double(s), length = len),
         xlam = rep(as.double(xlam), length = len),
       status = rep(as.integer(status), length = len),
        bound = rep(as.double(bound), length = len),
          len = as.integer(len)
  , PACKAGE = "dcdflib")
}

cdftV <- function(which, p, q, t, df, status, bound)
{
  len <- max(length(p), length(q), length(t), length(df), length(status), length(bound))
 .C("V_cdft",
        which = as.integer(which),
            p = rep(as.double(p), length = len),
            q = rep(as.double(q), length = len),
            t = rep(as.double(t), length = len),
           df = rep(as.double(df), length = len),
       status = rep(as.integer(status), length = len),
        bound = rep(as.double(bound), length = len),
          len = as.integer(len)
  , PACKAGE = "dcdflib")
}

cdftncV <- function(which, p, q, t, df, pnonc, status, bound)
{
  len <- max(length(p), length(q), length(t), length(df), length(pnonc), length(status), length(bound))
 .C("V_cdftnc",
        which = as.integer(which),
            p = rep(as.double(p), length = len),
            q = rep(as.double(q), length = len),
            t = rep(as.double(t), length = len),
           df = rep(as.double(df), length = len),
        pnonc = rep(as.double(pnonc), length = len),
       status = rep(as.integer(status), length = len),
        bound = rep(as.double(bound), length = len),
          len = as.integer(len)
  , PACKAGE = "dcdflib")
}

